zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/controller/usermanagement/AuthorizationController.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.gateway.rest.controller.usermanagement;
+
+import io.camunda.service.AuthorizationServices;
+import io.camunda.zeebe.gateway.rest.RequestMapper;
+import io.camunda.zeebe.gateway.rest.controller.CamundaRestController;
+import io.camunda.zeebe.gateway.rest.controller.usermanagement.dto.AuthorizationAssignRequest;
+import io.camunda.zeebe.protocol.impl.record.value.authorization.AuthorizationRecord;
+import java.util.concurrent.CompletableFuture;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+@CamundaRestController
+@RequestMapping("/v2/authorizations")
+public class AuthorizationController {
+  private final AuthorizationServices<AuthorizationRecord> authorizationServices;
+
+  public AuthorizationController(
+      final AuthorizationServices<AuthorizationRecord> authorizationServices) {
+    this.authorizationServices = authorizationServices;
+  }
+
+  @PostMapping(
+      produces = {MediaType.APPLICATION_JSON_VALUE, MediaType.APPLICATION_PROBLEM_JSON_VALUE},
+      consumes = MediaType.APPLICATION_JSON_VALUE)
+  public CompletableFuture<ResponseEntity<Object>> createAuthorization(
+      @RequestBody final AuthorizationAssignRequest authorizationAssignRequest) {
+    return RequestMapper.executeServiceMethodWithNoContentResult(
+        () ->
+            authorizationServices
+                .withAuthentication(RequestMapper.getAuthentication())
+                .createAuthorization(
+                    authorizationAssignRequest.getOwnerKey(),
+                    authorizationAssignRequest.getOwnerType(),
+                    authorizationAssignRequest.getResourceKey(),
+                    authorizationAssignRequest.getResourceType(),
+                    authorizationAssignRequest.getPermissions()));
+  }
```

Comment:
❌ We should change this slightly. I think I may have missed this in the create user PR as well as I never touched the REST API before I reviewed this.

The request that comes in is a dto. We should use the `RequestMapper` to map it to an internal object. Similar to what we do with other endpoints that have a request body. In this `RequestMapper` we can do some validations to check if the incoming request is valid. I'm not sure if there's specific validations that make sense here. Maybe the permissions need to adhere to a specific format?